### ISSUE TITLE
Code field in the node

### ISSUE THREAD

**Issue by p33k** [insolite/node-red-contrib-rcswitch3#1]
I'm not sure how to use the code field.

on the rcswitch2 node I triggered it with boolean true or false. and had this options:
"pin": 0,
"device": "C",
"group": "01000",

How do I use the code field in your node? I tried "01000",3 but that doesn't work

**Comment by slash-oleh** [comment]
Which code exactly do you trying to send? Note that pin/device/group fields are just inputs to handy raw code generators for specific devices. So the final raw code for `pin:0, device: C, group 01000` will be longer than just `01000`. The "code" node input supports only [raw binary strings](https://github.com/insolite/node-rcswitch2#rcswitchsendcode) at the moment. Also pay attention to bit count as it does matter. For example `01000` and `0001000` has different behavior. If you put a value of number type into node input (like `2643`) then it will be converted to binary string with 24 zeros padding by default (`"00..00101001010011"` - 24 bits).

**Comment by p33k** [comment]
I have a remote like "type a" in your manual.
in the rcswitch2 node I just had to put in the pin (0) the device (C) and the group (01000) in my case.
https://camo.githubusercontent.com/24335d7b721529a1e53922a97b60c564d9781c65/68747470733a2f2f7261772e6769746875622e636f6d2f696e736f6c6974652f6e6f64652d7263737769746368322f6d61737465722f696d672f747970655f612e6a7067
so what exactly do I have to put in the code field of your node? (for example for the one in the picture)

**Comment by slash-oleh** [comment]
TL;DR: If I calculated correctly then it's `4215889` for `on` state and `4215892` for `off` state with 24 bits padding or:
on: `"010000000101010001010001"`
off: `"010000000101010001010100"`
You can trace the algorithm here:
[switchOn](https://github.com/insolite/node-rcswitch2/blob/master/externals/rcswitch-pi/RCSwitch.cpp#L277)
[getCodeWordA](https://github.com/insolite/node-rcswitch2/blob/master/externals/rcswitch-pi/RCSwitch.cpp#L296)
[sendTriState](https://github.com/insolite/node-rcswitch2/blob/master/externals/rcswitch-pi/RCSwitch.cpp#L443)

Actually I've never faced that types of devices IRL as well as didn't document them :slightly_smiling_face:. Code sending docs were mostly just forked from original `node-rcswitch` package. Then I've simplified the `node-red` node to make it possible to send raw byte strings instead which covers all the cases comparing to original version which forces group/switch/etc. using.

**Comment by p33k** [comment]
Ok, for my workflow with the type of remotes I have its way easier with the rcswitch2 node.
So I just installed that also. 
Now I can use your receive node and rcswitch2 for sending
